src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -131,6 +131,54 @@ class Target
             await VerifyAnalyzerAsyncCs(source, tfmAndOption);
         }
 
+        [Fact]
+        public async Task MethodsWithOsDependentTypeParameterWarns()
+        {
+            var csSource = @"
+using System.Runtime.Versioning;
+
+[SupportedOSPlatform(""windows"")]
+class WindowsOnlyType {}
+
+public class Test
+{
+    void GenericMethod<T>() {}
+    void GenericMethod2<T1, T2>() {}
+    void M1()
+    {
+        [|GenericMethod<WindowsOnlyType>()|];
+        [|GenericMethod2<Test, WindowsOnlyType>()|];
+    }
+}
+" + MockAttributesCsSource;
+            await VerifyAnalyzerAsyncCs(csSource);
+        }
+
+        [Fact]
+        public async Task ConstructorWithOsDependentTypeParameterWarns()
+        {
+            var csSource = @"
+using System.Runtime.Versioning;
+
+[SupportedOSPlatform(""windows"")]
+class WindowsOnlyType {}
+
+class GenericClass<T> {}
```

Comment:
Can we add a test such as below:

```csharp
using System.Runtime.Versioning;

class GenericClass<T>
{
}

[SupportedOSPlatform("windows")]
class WindowsOnlyType { }

class DerivedWindowsOnlyType : GenericClass<WindowsOnlyType> { }
class Test
{
    void M<T>()
    {
        var t = new DerivedWindowsOnlyType();  // Should be flagged
    }
}
```

Reply:
Another test suggestion:

```csharp
using System.Runtime.Versioning;

class GenericClass<T>
{
}

[SupportedOSPlatform("windows")]
class WindowsOnlyType { }

class Test<T>
    where T : WindowsOnlyType
{
    void M<T>()
    {
        var t = new GenericClass<T>(); // Should be flagged.
    }
}
```

Reply:
I am not sure it could be flagged, i will try

Reply:
I think these scenarios related to the attribute's inheritance and the platform attributes defined as  [`[AttributeUsage(AttributeTargets.Assembly | ... , Inherited = false)]` ](https://github.com/dotnet/runtime/blob/b45ee9d37afec0c88141053e86ccf71c6f283000/src/libraries/System.Private.CoreLib/src/System/Runtime/Versioning/PlatformAttributes.cs#L63). So I don't think these scenarios should be covered/flagged correct me if i am wrong @terrajobst

Reply:
We might want to warn in case the base constructor is doing any platform-specific operation. Created separate issue for discussing and further handling this scenario https://github.com/dotnet/roslyn-analyzers/issues/4404